h2o-r/h2o-package/R/communication.R

```diff
@@ -799,9 +799,42 @@ h2o.list_jobs <- function() {
   df
 }
 
-h2o.get_job <- function(job_key) {
+h2o.get_job <- function(job_key, jobPoolSucees = FALSE, jobIsRecoverable = FALSE) {
   myJobUrlSuffix <- paste0(.h2o.__JOBS, "/", job_key)
-  rawResponse <- .h2o.doSafeGET(urlSuffix = myJobUrlSuffix)
+
+  # If request fail, repeat the request unless it is stopifnot(..) error or job is no longer exists in the cluster (e.g. in case of restart)
+  i <- 0
+  while (i < 30) {
+    rawResponse <- try(.h2o.doSafeGET(urlSuffix = myJobUrlSuffix))
+    if(class(rawResponse) == "try-error" && jobPoolSucees){
+      error_type <- attr(rawResponse,"condition")
+      if (jobIsRecoverable) {
+        print(sprintf("Job request failed %s, waiting for cluster to restart.", error_type$message))
+        Sys.sleep(10)
+      } else {
+        if(grepl("Job is missing", error_type$message, fixed = TRUE)) {
```

Comment:
Interesting, I am not even sure if the python client handles this case - this might be a bug in Python client. Good thinking!

However, I do think we need to detect the state in another way: grepping the message seems fragile.

Reply:
We conclude on a call that it is fragile but it is a best effort to skip unnecessary repeat call. In case of this check failure (e.g because of API change), the implementation is still working analogically to python client.